pkg/ingester/client/client.go

```diff
@@ -47,8 +47,14 @@ func MakeIngesterClient(addr string, cfg Config) (HealthAndIngesterClient, error
 	if err != nil {
 		return nil, err
 	}
+
+	ingClient := NewIngesterClient(conn)
+	if cfg.WriteRequestsBufferPoolingEnabled {
```

Comment:
I was debating this too myself. My idea was to avoid adding extra code on that path if pooling is disabled, but that may not be good enough reason. I'm ok dropping the config here.